### ISSUE TITLE
Define GraphQL schema for Basic User Identity

### ISSUE THREAD

**Issue by rudolfs** [radicle-dev/radicle-upstream#176]
### Create identity mutation
- given: user handle, display name, avatar url
- returns: user identity
  - user id
  - user handle
  - display name
  - avatar url
  - shareable entity identifier

### Project list query
  - given: user id
  - returns: list of projects
    - project name
    - project description
    - if project is registered (bool)
    - project stats
      - commit count
      - branch count
      - contributor count

**Comment by MeBrei** [comment]
I think for the Basic User Identity, the project query only lists the local projects, so that we don't have to pass the user id. The registered boolean is also not needed as only projects associated with orgs can be registered.

First sketch of the new involved schema changes:

```graphql
type Mutation {
  createUser(handle: String!, displayName: String, imgUrl: String): User!
  ...
}

type User {
  id: ID!
  handle: String!
  displayName: String
  imgUrl: String
  shareableEntityIdentifier: String!
}

type ProjectStats {
  commits: i32!
  branches: i32!
  contributors: i32!
}

"""Input object for project metadata"""
input MetadataInput {
  ...

  """Statistics for the project"""
  stats: ProjectStats!
}
```

**Comment by MeBrei** [comment]
Good point! Here is the updated schema:

```graphql
type Mutation {
  createUser(handle: String!, displayName: String, imgUrl: String): User!
  ...
}

type User {
  id: ID!
  handle: String!
  displayName: String
  imgUrl: String
  shareableEntityIdentifier: String!
}

"""Project Statistics"""
type ProjectStats {
  commits: Int!
  branches: Int!
  contributors: Int!
}

"""Radicle project"""
type Project {
  ...

  """Statistics for the project"""
  stats: ProjectStats!
}
```

**Comment by NunoAlexandre** [comment]
@rudolfs sorry, I was just passing by and saw this `if project is registered (bool)`. Usually, opting for booleans is not the best thing to do. Consider having some type like `ProjectState = Registered | NotRegistered` instead. Shall you need something more than a binary state, you can extend it at ease and with the types describing it clearly.

_flyes away_

**Comment by rudolfs** [comment]
@NunoAlexandre sounds good!

**Comment by rudolfs** [comment]
@NunoAlexandre I had another look at the designs, and we're only using this to either show a "registered" badge to the user or not. So I think packing this into an enum is just going to make more work on both back- and front-end, with no added benefit.

**Comment by NunoAlexandre** [comment]
@rudolfs that's fine, it's all about trade-offs and you figured yours here 👍

**Comment by rudolfs** [comment]
Implemented in #199
